File: core/src/net/bossdragon/component/Player.java
package net.bossdragon.component;

import com.artemis.Component;
import net.bossdragon.util.ActionTimer;

/**
 *
 */
public class Player extends Component {
    public final ActionTimer slide = new ActionTimer(C.Player.SlideDuration);
    public final ActionTimer slideCooldown = new ActionTimer(C.Player.SlideCooldownDuration);

    public int slideDirX = 0;
    public int slideDirY = 0;

    public int carriedSoldierId = -1;

    public boolean isSliding() {
        return slide.isRunning();
    }

    public boolean canStartSlide() {
        return !isSliding() && !slideCooldown.isRunning();
    }

    public boolean isCarryingAnySoldier() {
        return carriedSoldierId >= 0;
    }
}

File: core/src/net/bossdragon/component/Player.kt
package net.bossdragon.component

import com.artemis.Component
import com.badlogic.gdx.math.Vector2
import net.bossdragon.enums.C
import net.bossdragon.util.ActionTimer

class Player : Component() {
    val slide = ActionTimer(C.Player.SlideDuration)
    val slideCooldown = ActionTimer(C.Player.SlideCooldownDuration)

    var slideDirX = 0
    var slideDirY = 0

    val isSliding: Boolean
        get() = slide.isRunning

    val canStartSlide: Boolean
        get() = !isSliding && !slideCooldown.isRunning
}
